R-package/tests/testthat/test_basic.R

```diff
@@ -2773,7 +2773,7 @@ test_that(paste0("lgb.train() throws an informative error if the members of inte
     }, "every element in interaction_constraints must be a character vector or numeric vector")
 })
 
-test_that("lgb.train() throws an informative error if interaction_constraints contains a too large index", {
+test_that("lgb.train() throws an error if interaction_constraints contains wrong features", {
```

Comment:
Why are you proposing changing this test? I'd expect this behavior to still be true based on the PR description, and I'd prefer that we still check for *specific* errors in any `expect_error()` calls to be sure that tests don't accidentally pass when unexpected errors are raised.

Reply:
> Why are you proposing changing this test? I'd expect this behavior to still be true based on the PR description, and I'd prefer that we still check for _specific_ errors in any `expect_error()` calls to be sure that tests don't accidentally pass when unexpected errors are raised.

Right. I have reverted the wording, and added the explicit error message. It will be less specific because the actual check also handles the case of feature names.

Reply:
Now, I have changed the error message (and the test) so that it gives `"unknown features(s) in interaction_constraints: '127', '128'"`